File: src/main/java/BB/resto/Entity/ZoneLivraison.java
package BB.resto.Entity;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class ZoneLivraison {
    private int id;
    private String nom;
    private double prixLivraison;
    private String quartiersCouverts;
    private LocalDateTime createdAt;


    private List<String> quartiersList;


    public ZoneLivraison() {
        this.createdAt = LocalDateTime.now();
        this.quartiersList = new ArrayList<>();
    }

    public ZoneLivraison(String nom, double prixLivraison, String quartiersCouverts) {
        this();
        this.nom = nom;
        this.prixLivraison = prixLivraison;
        this.quartiersCouverts = quartiersCouverts;
        parseQuartiers();
    }


    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }

    public double getPrixLivraison() { return prixLivraison; }
    public void setPrixLivraison(double prixLivraison) { this.prixLivraison = prixLivraison; }

    public String getQuartiersCouverts() { return quartiersCouverts; }
    public void setQuartiersCouverts(String quartiersCouverts) {
        this.quartiersCouverts = quartiersCouverts;
        parseQuartiers();
    }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }


    private void parseQuartiers() {
        quartiersList = new ArrayList<>();
        if (quartiersCouverts != null && !quartiersCouverts.trim().isEmpty()) {
            String[] parts = quartiersCouverts.split(",");
            for (String part : parts) {
                quartiersList.add(part.trim());
            }
        }
    }

    public List<String> getQuartiersList() {
        if (quartiersList == null || quartiersList.isEmpty()) {
            parseQuartiers();
        }
        return quartiersList;
    }

    public void setQuartiersList(List<String> quartiersList) {
        this.quartiersList = quartiersList;
        this.quartiersCouverts = String.join(", ", quartiersList);
    }

    public boolean couvreQuartier(String quartier) {
        return getQuartiersList().stream()
                .anyMatch(q -> q.equalsIgnoreCase(quartier.trim()));
    }

    public String getFormattedPrixLivraison() {
        return String.format("%.2f FCFA", prixLivraison);
    }

    @Override
    public String toString() {
        return "ZoneLivraison{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", prixLivraison=" + prixLivraison +
                ", quartiersCouverts='" + quartiersCouverts + '\'' +
                '}';
    }
}

File: BBurgerSymfony/BBurgerSymfony/src/Entity/ZoneLivraison.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'zone_livraison')]
class ZoneLivraison
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 50)]
    private ?string $nom = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prixLivraison = '0';

    #[ORM\Column(type: 'text')]
    private ?string $quartiersCouverts = null;

    #[ORM\Column(type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;
        return $this;
    }

    public function getPrixLivraison(): ?string
    {
        return $this->prixLivraison;
    }

    public function setPrixLivraison(string $prixLivraison): self
    {
        $this->prixLivraison = $prixLivraison;
        return $this;
    }

    public function getQuartiersCouverts(): ?string
    {
        return $this->quartiersCouverts;
    }

    public function setQuartiersCouverts(string $quartiersCouverts): self
    {
        $this->quartiersCouverts = $quartiersCouverts;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function getFormattedPrixLivraison(): string
    {
        return number_format((float)$this->prixLivraison, 2, ',', ' ') . ' FCFA';
    }

    public function getQuartiersList(): array
    {
        if (!$this->quartiersCouverts) {
            return [];
        }
        return array_map('trim', explode(',', $this->quartiersCouverts));
    }
    
    
    public function couvreQuartier(string $quartier): bool
    {
        $quartiers = $this->getQuartiersList();
        return in_array(trim($quartier), $quartiers);
    }
    
    
    public function getQuartiersFormatted(): string
    {
        $quartiers = $this->getQuartiersList();
        return implode(', ', $quartiers);
    }
    
    
    public function __toString(): string
    {
        return $this->nom . ' (' . $this->getFormattedPrixLivraison() . ')';
    }
}